### PR TITLE
Refactored profs and added more tests.

### PR DESCRIPTION
**Pull request by sean-gilliam**
This PR refactored profs to be more C++ like.

The following changes are:

- Changed logic to support vectors.
- Documented most of the class methods.
- Fixed several tests that had faulty logic.
- Added more tests to support the refactoring.

The tests don't cover everything and I left a few of them to do, but it covers at least 80-85% of the class.

### CODE REVIEW

File: tests/prof_tests.c [hunk 1]
```diff
@@ -1,96 +1,1339 @@
...
+				auto result = player->Profs()->HasProf(6);
+
+				REQUIRE(result == false);
+			}
+
+			TestHelperCleanupPlayerObject(player);
+		}
+	}
+
+	GIVEN("a character who doesn't have the proficiency")
```
Comment (rezalas) [inline tests/prof_tests.c:127]: This should be "has the proficiency" I think

File: tests/prof_tests.c [hunk 2]
```diff
@@ -1,96 +1,1339 @@
...
+	{
+		WHEN("the function is called")
+		{
+			auto player = new char_data();
+			player->Profs()->SetChar(player);
+			player->Profs()->SetProf(6, 1); // cooking
+
+			THEN("it should return false")
```
Comment (rezalas) [inline tests/prof_tests.c:135]: Checking for true in the body but says false

Comment (sean-gilliam) [reply tests/prof_tests.c:135]: copypasta strikes again

**Review by rezalas — APPROVED** [review]
LGTM!!

**Pull request merged**